feat: initial environment variable provider (#239)
open-feature/js-sdk-contrib: libs/providers/env-var/README.md

# Environment Variable Provider

The environment variable provider is a great way to start using OpenFeature.
It doesn't require any infrastructure to setup or manage, and provides a simple way to gain experience with the core concepts of feature flagging.
However, it doesn't support features such as dynamic updates at run-time or contextual flag evaluation.
That's where feature flags become extremely powerful!
Thankfully, the OpenFeature SDK supports basic providers such at this one, while making it simple to switch to a more powerful system when the time is right.

## Installation

```
$ npm install @openfeature/env-var-provider
```

Required peer dependencies

```
$ npm install @openfeature/js-sdk
```

## Usage

> The environment variable provider uses the environment variables defined on the application to determine the value of a feature flag.

Reviewer: ```suggestion
The environment variable provider uses the environment variables to determine the value of a feature flag.
```

> It supports `booleans`, `strings`, `numbers` and `objects` by attempting to cast the value of an environment variable to the requested type.

Reviewer: ```suggestion
It supports `booleans`, `strings`, `numbers` and `objects` by attempting to interpret the value of an environment variable to the requested type.
```

> The fallback value will be return if the environment variable doesn't exist or the value can't be cast to the desired type.

Reviewer: ```suggestion
The default value will be returned if the environment variable doesn't exist or the value can't be cast to the desired type.
```
